tests/governance_js.py

```diff
@@ -507,22 +508,43 @@ def test_apply(network, args):
             "/gov/proposals",
             proposal(action("always_throw_in_apply")),
         )
-        assert r.status_code == 200, r.body.text()
-        assert r.body.json()["state"] == "Failed", r.body.json()
+        assert r.status_code == 500, r.body.text()
+        assert r.body.json()["error"]["code"] == "InternalError", r.body.json()
         assert (
-            r.body.json()["failure"]["reason"]
+            r.body.json()["error"]["message"].split("\n")[0]
             == "Failed to apply(): Error: Error message"
         ), r.body.json()
 
+    with node.client(None, "member0") as c:
+        pprint.pprint(
+            proposal(action("always_accept_noop"), action("always_throw_in_apply"))
+        )
+        r = c.post(
+            "/gov/proposals",
+            proposal(action("always_accept_noop"), action("always_throw_in_apply")),
+        )
+        assert r.status_code == 200, r.body().text()
+        proposal_id = r.body.json()["proposal_id"]
+        r = c.post(f"/gov/proposals/{proposal_id}/ballots", ballot_yes)
+        assert r.status_code == 200, r.body().text()
+
+        with node.client(None, "member1") as c:
+            r = c.post(f"/gov/proposals/{proposal_id}/ballots", ballot_yes)
+            assert r.body.json()["error"]["code"] == "InternalError", r.body.json()
+            assert (
+                r.body.json()["error"]["message"].split("\n")[0]
+                == "Failed to apply(): Error: Error message"
+            ), r.body.json()
```

Comment:
Nit: I'm wary of precise asserts on error message format like this. I'd be tempted to say `assert(error_message.contains("Failed to apply()"))` and `assert(error_message.contains("Apply message"))` (and the same below), though in this case that's very similar, so not critical.

Reply:
Can I ask you why you are wary of being precise here? I think I'd like to know if this changes (accidentally or not) in the future.

Reply:
I just think it's likely that at some point this does change, we introduce a newline after `apply():` for instance, and this assertion fails, and after staring at callstacks for a while we update the asserts to be `.contains()`. Making the change in advance gives us a little bit of flexibility, saving us time for that kind of incidental restructuring. We're asserting that the error message contains the things we think are important, but we're not strict in structure.

Reply:
I'd bet that day never comes, Now we'll miss changes that bork the line returns and we'll never know, but ok, I've done it.